bolt-sidecar/src/config/signing.rs

```diff
@@ -1,29 +1,133 @@
-    pub(super) private_key: Option<String>,
-    /// URL for the commit-boost sidecar
-    #[clap(
-        long,
-        env = "BOLT_SIDECAR_CB_SIGNER_URL",
-        conflicts_with("private_key"),
-        requires("commit_boost_jwt_hex")
-    )]
-    pub(super) commit_boost_url: Option<String>,
+    #[clap(long, env = "BOLT_SIDECAR_PRIVATE_KEY")]
+    pub private_key: Option<BlsSecretKey>,
+    /// Socket address for the commit-boost sidecar
+    #[clap(long, env = "BOLT_SIDECAR_CB_SIGNER_URL", requires("commit_boost_jwt_hex"))]
+    pub commit_boost_address: Option<SocketAddr>,
     /// JWT in hexadecimal format for authenticating with the commit-boost service
-    #[clap(
-        long,
-        env = "BOLT_SIDECAR_CB_JWT_HEX",
-        conflicts_with("private_key"),
-        requires("commit_boost_url")
-    )]
-    pub(super) commit_boost_jwt_hex: Option<String>,
+    #[clap(long, env = "BOLT_SIDECAR_CB_JWT_HEX", requires("commit_boost_address"))]
+    pub commit_boost_jwt_hex: Option<JwtSecretConfig>,
+    /// The password for the ERC-2335 keystore.
+    /// Reference: https://eips.ethereum.org/EIPS/eip-2335
+    #[clap(long, env = "BOLT_SIDECAR_KEYSTORE_PASSWORD")]
+    pub keystore_password: Option<ZeroizeString>,
+}
+
+// Implement Debug manually to hide the keystore_password field
+impl fmt::Debug for SigningOpts {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SigningOpts")
+            .field("private_key", &self.private_key)
+            .field("commit_boost_url", &self.commit_boost_address)
+            .field("commit_boost_jwt_hex", &self.commit_boost_jwt_hex)
+            .field("keystore_password", &"********") // Hides the actual password
+            .finish()
+    }
+}
+
+impl Default for SigningOpts {
+    fn default() -> Self {
+        Self {
+            private_key: Some(BlsSecretKey::random_bls_secret()),
+            commit_boost_address: None,
+            commit_boost_jwt_hex: None,
+            keystore_password: None,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct BlsSecretKey(pub SecretKey);
+
+impl BlsSecretKey {
+    pub fn random_bls_secret() -> Self {
+        let mut rng = rand::thread_rng();
+        let mut ikm = [0u8; 32];
+        rng.fill_bytes(&mut ikm);
+        Self(SecretKey::key_gen(&ikm, &[]).unwrap())
+    }
+}
+
+impl From<&str> for BlsSecretKey {
+    fn from(sk: &str) -> Self {
+        let hex_sk = sk.strip_prefix("0x").unwrap_or(sk);
+        let sk = SecretKey::from_bytes(&hex::decode(hex_sk).expect("valid hex")).expect("valid sk");
+        BlsSecretKey(sk)
+    }
+}
+
+impl Deref for BlsSecretKey {
+    type Target = SecretKey;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl fmt::Display for BlsSecretKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "0x{}", hex::encode(self.0.to_bytes()))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct JwtSecretConfig(pub String);
```

Comment:
Yes I've checked it before implementing it and it is a wrapper over `[u8; 32]` but the underlying array is not accessible, and it doesn't implement `ToString` or `Default`. As such I could not transform it in a hex string as needed by Commit-Boost.

Reply:
in alloy it has `as_bytes` but still not sure if worth to use it. maybe take a look https://github.com/alloy-rs/alloy/blob/ef9b58b617e33eba326a219a4dfb705b72db0dde/crates/rpc-types-engine/src/jwt.rs#L266

Reply:
Oh I see, haven't found it because it was implemented recently (2024-09-25) with this commit: https://github.com/alloy-rs/alloy/commit/6ff6a9d91a7004f529eb1f2dc6735e7d3420ef69